src/main/java/edu/cornell/kfs/module/purap/document/web/struts/CuPurchaseOrderAction.java

```diff
@@ -162,4 +174,57 @@ public ActionForward cancel(ActionMapping mapping, ActionForm form,
         return forward;
     }
 
+    /**
+     * Overridden to perform the proper account-line-override setup
+     * on POA items' account addLines, as well as on all of the items'
+     * existing accounting lines instead of just the filtered list
+     * from the document's getSourceAccountingLines() method override.
+     * 
+     * @see org.kuali.kfs.sys.web.struts.KualiAccountingDocumentActionBase#processAccountingLineOverrides(
+     * org.kuali.kfs.sys.web.struts.KualiAccountingDocumentFormBase)
+     */
+    @Override
+    protected void processAccountingLineOverrides(KualiAccountingDocumentFormBase transForm) {
+        if (transForm.hasDocumentId() && transForm.getDocument() instanceof PurchaseOrderAmendmentDocument) {
+            PurchaseOrderAmendmentDocument poaDocument = (PurchaseOrderAmendmentDocument) transForm.getDocument();
+            processAccountingLineOverrides(transForm.getNewSourceLine());
+            processAccountingLineOverrides(transForm.getNewTargetLine());
+            processAccountingLineOverrides(poaDocument, getItemSourceAccountingLines(poaDocument));
+            processAccountingLineOverrides(poaDocument, poaDocument.getTargetAccountingLines());
+            /*
+             * Using null document as first arg, for consistency with regular source/target addLine handling
+             * in KualiAccountingDocumentActionBase.
+             */
+            processAccountingLineOverrides(null, getItemAccountingAddLines(poaDocument));
+        } else {
+            super.processAccountingLineOverrides(transForm);
+        }
+    }
+
+    protected List<PurApAccountingLine> getItemSourceAccountingLines(PurchaseOrderAmendmentDocument document) {
+        return getAccountingLinesFromItems(document,
+                (itemStream) -> itemStream.flatMap(this::getItemSourceAccountingLinesAsStream));
```

Comment:
can this lambda expression and the similar ones below be refactored to help with understanding, maybe some different naming.

Reply:
Ok, I will refactor this accordingly to improve readability.